// slate-transcript-editor/components/SideBtns/index.js
import React, { useState, useEffect } from 'react';
import Divider from '@material-ui/core/Divider';
import Tooltip from '@material-ui/core/Tooltip';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import Menu from '@material-ui/core/Menu';
import Link from '@material-ui/core/Link';
import Grid from '@material-ui/core/Grid';
import MenuItem from '@material-ui/core/MenuItem';
import SaveAltIcon from '@material-ui/icons/SaveAlt';
import SaveOutlinedIcon from '@material-ui/icons/SaveOutlined';
import KeyboardReturnOutlinedIcon from '@material-ui/icons/KeyboardReturnOutlined';
import MusicNoteOutlinedIcon from '@material-ui/icons/MusicNoteOutlined';
import HelpOutlineOutlinedIcon from '@material-ui/icons/HelpOutlineOutlined';
import ImportExportIcon from '@material-ui/icons/ImportExport';
import CachedOutlinedIcon from '@material-ui/icons/CachedOutlined';
import InfoOutlined from '@material-ui/icons/InfoOutlined';
import KeyboardArrowDownIcon from '@material-ui/icons/KeyboardArrowDown';
import RedoIcon from '@material-ui/icons/Redo';
import UndoOutlinedIcon from '@material-ui/icons/UndoOutlined';
import EmojiSymbolsOutlinedIcon from '@material-ui/icons/EmojiSymbolsOutlined';
import subtitlesExportOptionsList from '../../util/export-adapters/subtitles-generator/list.js';

function SideBtns({
  handleExport,
  isProcessing,
  isContentModified,
  isContentSaved,
  setIsProcessing,
  insertTextInaudible,
  handleInsertMusicNote,
  handleSplitParagraph,
  handleRestoreTimecodes,
  handleReplaceText,
  handleSave,
  handleAnalyticsEvents,
  REPLACE_WHOLE_TEXT_INSTRUCTION,
  optionalBtns,
  handleUndo,
  handleRedo,
  isEditable
}) {
  const [anchorMenuEl, setAnchorMenuEl] = useState(null); // used by MUI export menu

  const handleMenuClose = () => {
    setAnchorMenuEl(null);
  }; // used by MUI export menu


  const handleMenuClick = event => {
    setAnchorMenuEl(event.currentTarget);
  };

  return /*#__PURE__*/React.createElement(Grid, {
    container: true,
    direction: "column",
    justify: "flex-start",
    alignItems: "stretch"
  }, /*#__PURE__*/React.createElement(Grid, {
    item: true
  }, /*#__PURE__*/React.createElement(Tooltip, {
    title: /*#__PURE__*/React.createElement(Typography, {
      variant: "body1"
    }, "Export options")
  }, /*#__PURE__*/React.createElement(Button, {
    "aria-controls": "simple-menu",
    "aria-haspopup": "true",
    onClick: handleMenuClick
  }, /*#__PURE__*/React.createElement(SaveAltIcon, {
    color: "primary"
  }), " ", /*#__PURE__*/React.createElement(KeyboardArrowDownIcon, {
    color: "primary"
  }))), /*#__PURE__*/React.createElement(Menu, {
    id: "simple-menu",
    anchorEl: anchorMenuEl,
    keepMounted: true,
    open: Boolean(anchorMenuEl),
    onClose: handleMenuClose
  }, /*#__PURE__*/React.createElement(Divider, null), /*#__PURE__*/React.createElement(MenuItem, {
    onClick: handleMenuClose,
    disabled: true
  }, /*#__PURE__*/React.createElement(Link, {
    style: {
      color: 'black'
    }
  }, "Closed Captions Export")), subtitlesExportOptionsList.map(({
    type,
    label,
    ext
  }, index) => {
    return /*#__PURE__*/React.createElement(MenuItem, {
      key: index + label,
      onClick: () => {
        handleExport({
          type,
          ext,
          isDownload: true
        });
        handleMenuClose();
      }
    }, /*#__PURE__*/React.createElement(Link, {
      color: "primary"
    }, label, " (", /*#__PURE__*/React.createElement("code", null, ".", ext), ")"));
  }), /*#__PURE__*//*React.createElement(Divider, null), /*#__PURE__*//*React.createElement(MenuItem, {
    onClick: handleMenuClose,
    disabled: true
  }, /*#__PURE__*//*React.createElement(Link, {
    style: {
      color: 'black'
    }
  }, "Developer options")), /*#__PURE__*//*React.createElement(MenuItem, {
    onClick: () => {
      handleExport({
        type: 'json-slate',
        ext: 'json',
        speakers: true,
        timecodes: true,
        isDownload: true
      });
      handleMenuClose();
    }
  }, /*#__PURE__*//*React.createElement(Link, {
    color: "primary"
  }, "SlateJs (", /*#__PURE__*//*React.createElement("code", null, ".json"), ")")), /*#__PURE__*//*React.createElement(MenuItem, {
    onClick: () => {
      handleExport({
        type: 'json-digitalpaperedit',
        ext: 'json',
        speakers: true,
        timecodes: true,
        isDownload: true
      });
      handleMenuClose();
    }
  }, /*#__PURE__*//*React.createElement(Link, {
    color: "primary"
  }, "DPE (", /*#__PURE__*//*React.createElement("code", null, ".json"), ")"))*/)),/*#__PURE__*/React.createElement(Grid, {
    item: true
  }, /*#__PURE__*/React.createElement("br", null)), /*#__PURE__*/React.createElement(Grid, {
    item: true
  }, /*#__PURE__*/React.createElement(Tooltip, {
    title: /*#__PURE__*/React.createElement(Typography, {
      variant: "body1"
    }, "Undo ", /*#__PURE__*/React.createElement("br", null), /*#__PURE__*/React.createElement("code", null, "cmd"), " ", /*#__PURE__*/React.createElement("code", null, "z"))
  }, /*#__PURE__*/React.createElement(Button, {
    onClick: handleUndo,
    color: "primary"
  }, /*#__PURE__*/React.createElement(UndoOutlinedIcon, {
    color: "primary"
  }))), /*#__PURE__*/React.createElement(Tooltip, {
    title: /*#__PURE__*/React.createElement(Typography, {
      variant: "body1"
    }, "Redo ", /*#__PURE__*/React.createElement("br", null), " ", /*#__PURE__*/React.createElement("code", null, "cmd"), " ", /*#__PURE__*/React.createElement("code", null, "shift"), " ", /*#__PURE__*/React.createElement("code", null, "z"))
  }, /*#__PURE__*/React.createElement(Button, {
    onClick: handleRedo,
    color: "primary"
  }, /*#__PURE__*/React.createElement(RedoIcon, {
    color: "primary"
  })))), /*#__PURE__*/React.createElement(Grid, {
    item: true
  }, /*#__PURE__*/React.createElement("br", null)), /*#__PURE__*/React.createElement(Grid, {
    item: true
  }, optionalBtns));
}

export default SideBtns;
